### PR TITLE
Making monstermos waaaay faster

### PR DESCRIPTION
**Pull request by Putnam3145**
########################################
########################################
# If this is your first PR, or not, take the time to read our CONTRIBUTING.md file! This will ensure your PR is up to standards and get it merged more quickly! You can see it here: https://github.com/yogstation13/Yogstation/blob/master/.github/CONTRIBUTING.md
########################################
########################################

### Intent of your Pull Request

Makes monstermos about 10x as fast by keeping all the gas mixtures in a vector, which improves cache locality; doing all active turfs/equalize/reaction processing in C++, which also improves cache locality; and allowing the optimizer to optimize more aggressively, which improves performance a good chunk.

I only tested this with my basic plasma flood test because I couldn't get movement controls to work, good lord. Test this thoroughly before merging.

#### Changelog

:cl:  Putnam
tweak: Monstermos is now ultra fast, an upgrade from the previous super fast.
/:cl:

### CODE REVIEW

**Comment by monster860** [comment]
Before test merging it we need the linux build because the server runs linux.

Also PR is here: https://github.com/yogstation13/extools/pull/6/files

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -245,7 +245,7 @@
 		update |= CANISTER_UPDATE_HOLDING
 	if(connected_port)
 		update |= CANISTER_UPDATE_CONNECTED
-	var/pressure = air_contents.return_pressure()
+	var/pressure = air_contents ? air_contents.return_pressure() : 0
```
Comment (JamieD1) [inline code/modules/atmospherics/machinery/portable/canister.dm]:
```suggestion
	var/pressure = air_contents.return_pressure()
```

Comment (JamieD1) [reply code/modules/atmospherics/machinery/portable/canister.dm]: Reverting for test merge conflicts

**Comment by ReddicusDragon** [comment]
we need this dont give up

**Comment by Amelia0010** [comment]
@Putnam3145 @monster860 Status?

**Comment by maxion12345** [comment]
Ok

**Pull request closed without merging**